sparse_strips/vello_api/src/paint.rs

```diff
@@ -3,24 +3,51 @@
 
 //! Types for paints.
 
-use peniko::color::{AlphaColor, Srgb};
+use peniko::color::{AlphaColor, PremulRgba8, Srgb};
+
+/// A paint that needs to be resolved via its index.
+// In the future, we might add additional flags, that's why we have
+// this thin wrapper around u32, so we can change the underlying
+// representation without breaking the API.
+#[derive(Debug, Clone)]
+pub struct IndexedPaint(u32);
+
+impl IndexedPaint {
+    /// Create a new indexed paint from an index.
+    pub fn new(index: usize) -> Self {
+        assert!(
+            index <= u32::MAX as usize,
+            "exceeded the maximum number of paints"
+        );
+
+        #[allow(
+            clippy::cast_possible_truncation,
+            reason = "Guaranteed to be in-bounds"
+        )]
+        Self(index as u32)
```

Comment:
Nit: We should be able to use https://doc.rust-lang.org/stable/std/primitive.u32.html#impl-TryFrom%3Cusize%3E-for-u32 here